### ISSUE TITLE
e28-api - array in a field?

### ISSUE THREAD

**Issue by adejeffchen** [susanBuck/e28-fall20#43]
I have a resource that represents recipe. 
One of the fields is the list of ingredients. 
I notice I couldn't set up the field as an array so I set it up as type text. 
Then when I seed it with an array, the field became empty after seeded. (probably expected) 

Seed field example
```
"ingredients": [
                "2 tablespoons canola mayonnaise",
                "1 tablespoon 2% reduced-fat milk",
                "1 tablespoon butter, melted",
                "8 whole-grain slider buns, split and toasted"
            ],
```

So I was thinking I have two options..
In real world example, it might make sense to have a separate table to store [recipe ID, ingredient ID, ingredient text]
OR 
I can store ingredients as text with a | separator
`"ingredients": "2 tablespoons canola mayonnaise | 1 tablespoon 2% reduced-fat milk | 1 tablespoon butter, melted | 8 whole-grain slider buns, split and toasted", `

Are both okay? or is there a way to store an array in a field?

**Comment by susanBuck** [comment]
@adejeffchen -

The relational DB system powering the API does not support array data. As you noted, the real-world/ideal solution here would be to store the ingredients in a separate resource/table.

Or, as you also noted, you could encode the enumerated data as a string with some separator (`|`, `,`, etc.) and explode that string into an array upon retrieval. Not best DB practices, but fine enough for our simple examples.

In summary - either option would be okay.

**Comment by adejeffchen** [comment]
Thank you!
